src/Mouf/Database/TDBM/TDBMService.php

```diff
@@ -530,6 +530,39 @@ public function deleteObject(TDBMObject $object) {
 		}
 	}
 
+    /**
+     * This function removes the given object from the database. It will also remove all objects relied to the one given
+     * by parameter before all.
+     *
+     * Notice: if the object has a multiple primary key, the function will not work.
+     *
+     * @param TDBMObject $objToDelete
+     */
+    public function deleteCascade(TDBMObject $objToDelete) {
+        $this->deleteAllConstraintWithThisObject($objToDelete);
+        $this->deleteObject($objToDelete);
+    }
+
+    /**
+     * This function is used only in TDBMService (private function)
+     * It will call deleteCascade function foreach object relied with a foreign key to the object given by parameter
+     *
+     * @param TDBMObject $obj
+     * @return TDBMObjectArray
+     */
+    private function deleteAllConstraintWithThisObject(TDBMObject $obj) {
+        $table = $obj->_getDbTableName();
+        $constraints = $this->dbConnection->getConstraintsFromTable($table);
+        foreach ($constraints as $constraint) {
+            $sql = "SELECT DISTINCT ".$constraint["table1"].".* FROM ".$table." LEFT JOIN ".$constraint["table1"]." ON ".$table.".".$constraint["col2"]." = ".$constraint["table1"].".".$constraint["col1"]
```

Comment:
Column or table names should be protected with a call to `$this->dbConnection->escapeDBItem()`.
This way, if a table or a column has a name of a reserved keyword (like "order" for instance), TDBM keeps running smoothly.